src/var.c

```diff
@@ -166,10 +166,17 @@ int DEFAULT_SYMVER_PRE(fi_var_register)(const struct fi_provider *provider,
 	v->provider = provider;
 	v->var_name = strdup(var_name);
 	v->help_string = strdup(help_string);
-	if (!v->var_name || !v->help_string || 
-		asprintf(&v->env_var_name, "FI_%s_%s",
+
+	// core providers don't get core prefix
+	if (strncmp("core", provider->name, sizeof ("core")))
```

Comment:
This should just compare the provider pointers rather than doing a string compare